app/controllers/dashboard/advisers_controller.rb

```diff
@@ -7,10 +7,20 @@ def index
       @trading_names = @firm.trading_names
     end
 
+    def new
+      @firm = Firm.find(params[:firm_id])
+      @adviser = advisers.build
+    end
+
     private
 
     def principal
       current_user.principal
     end
+
+    def advisers
+      Firm.find_by(id: params[:firm_id], fca_number: principal.fca_number)
+        .advisers
```

Comment:
So, `principal.firm.trading_names.find_by(id: params[:id]).advisers` does the trick?